src/modules/job-list/match.c

```diff
@@ -428,6 +430,90 @@ static struct list_constraint *create_results_constraint (struct match_ctx *mctx
                                       errp);
 }
 
+static int match_hostlist (struct list_constraint *c,
+                           const struct job *job,
+                           unsigned int *comparisons,
+                           flux_error_t *errp)
+{
+    struct hostlist *hl = zlistx_first (c->values);
+    const char *host;
+
+    if (inc_check_comparison (c->mctx, comparisons, errp) < 0)
+        return -1;
```

Comment:
The count of comparisons is incremented here and also for each host. Doesn't that mean one host comparison is counted twice?

Reply:
I think my logic was there was a comparison simply to see if we should bother to check hosts (i.e. if the job didn't run, we don't check anything, and that should count for at least 1 comparison).  But now that you mention it, perhaps that is not the most sensible way to think about it.  I'll remove the first check.